qml/pages/LoginPage.qml

```diff
@@ -21,42 +21,117 @@ Page {
 
         onAccessTokenFail: {
             console.log('access token failed')
-            webView.visible = true
         }
     }
 
-    SilicaWebView {
-        id: webView
-        anchors.fill: parent
-        url: page.startUrl + "&state=" + page.processId
+    Slack.AuthServer {
+        id: server
 
-        header: PageHeader {
-            title: "Sign in with Slack"
+        onResultUrlAvailable: {
+            authenticator.fetchAccessToken(url);
         }
+    }
 
-        onNavigationRequested: {
-            if (isReturnUrl(request.url)) {
-                visible = false
-                request.action = WebView.IgnoreRequest
+    Connections {
+        target: server
+        onListening: {
+            listening.text = (server.isListening() ? qsTr("Listening on port %1") : qsTr("Not listening on port %1")).arg(3000)
+        }
+    }
 
-                if (isSuccessUrl(request.url)) {
-                    authenticator.fetchAccessToken(request.url)
+    onStatusChanged: {
+        switch (status) {
+        case PageStatus.Active:
+            server.listen(3000);
+            Qt.openUrlExternally(page.startUrl + "&state=" + page.processId);
```

Comment:
This causes the web browser to be automatically opened when the page is first shown. Is that intended as default behaviour, where the alternative options are just for "advanced" users? Or would it be better to let the user to choose the right method for themselves?

Reply:
What would be your preference?
I actually have more than one browser so this pops a dialog to let me chose the browser. But now I realize that fresh installs will directly start the login sequence in the browser (while I am still able to dismiss that dialog).
This was a last minute 'great idea' that maybe this new screen is not user friendly. I am open to suggestions if this should be enabled or removed.

Reply:
Hmm I would say let's keep the screen (for debugging purposes if nothing else) and go to webview by default (feels like a better flow). Regarding the browser, have you tried whether this works when the user users an Android browser, like Firefox?

Reply:
No, I havent yet tried with an AD browser (can you register one as url handler?)
I will try to show the webview first and offer an "Advanced" pull-down menu

Reply:
Done, see latest commit @danvratil